### ISSUE TITLE
The problem of accuracy of Dnn_regression model using tflearn

### ISSUE THREAD

**Issue by Rra2** [tflearn/tflearn#1060]
Question 1.
I want to check the accuracy of the regression result window, not binary_acc. Because of this, only the binary_acc is displayed in the tensorboard graph, and it is displayed as 0 only.

Question 2.
I would like to apply the loss function to mean absolute percentage error instead of mse. I would like to ask for advice on the code that should be added to objectives.py.

Question 3.
If there is a theoretical error of deep running in my code, I would appreciate your advice.

Here is my code and the results:

# -*- coding: utf-8 -*-
from xlrd import open_workbook
from sklearn.model_selection import train_test_split
import tflearn
import numpy as np
import random
import tensorflow as tf

class EarlyStoppingCallback(tflearn.callbacks.Callback):
    def __init__(self, loss_threshold):
        self.loss_threshold = loss_threshold

    def on_epoch_end(self, training_state):
        if training_state.val_loss is None: return
        if training_state.val_loss < self.loss_threshold and training_state.val_loss!=0.:
            raise StopIteration

# Loss function
def mean_absolute_percentage_error(y_test, estimation_result):
    y_test, estimation_result = np.array(y_test), np.array(estimation_result)
    return np.mean(np.abs((y_test - estimation_result) / y_test)) * 100

# control variables
needed_col_numbers_for_data = [2,3,4,5,6,7,8,9,10,11]
num_feat = 9
needed_col_numbers_for_label = [11]
starting_row_index = 4  # this variable can be modified according to the data format in given excel file

# read data from excel file
filename = 'concrete_mix_data_6.xlsx'
workbook = open_workbook(filename)
worksheet = workbook.sheet_by_index(0)
nrows = worksheet.nrows
ncols = worksheet.ncols

# data preparation --> data & label
ending_row_index = nrows
data_and_label = []
for row_num in range(starting_row_index-1, ending_row_index):
    data_and_label.append([worksheet.row_values(row_num)[i] for i in needed_col_numbers_for_data])
random.shuffle(data_and_label)
data_and_label = np.array(data_and_label)
X = data_and_label[:,:-1]
Y = np.array(data_and_label[:,-1])
Y = np.reshape(Y, (len(Y),1))

# split data into training and testing group
x_train, x_test, y_train, y_test = train_test_split(X, Y, test_size=0.25)

# init
tflearn.init_graph(num_cores=4, gpu_memory_fraction=0.5)

print('======================= DNN regression model  ==================')

# Build neural network
net = tflearn.input_data(shape=[None, num_feat])
net = tflearn.fully_connected(net, 1000, activation='relu', name='Layer1')
net = tflearn.dropout(net, 0.5, name='Dropout1')
net = tflearn.fully_connected(net, 1000, activation='relu', name='Layer2')
net = tflearn.dropout(net, 0.5, name='Dropout2')
net = tflearn.fully_connected(net, 1000, activation='relu', name='Layer3')
net = tflearn.dropout(net, 0.5, name='Dropout3')
net = tflearn.fully_connected(net, 1000, activation='relu', name='Layer4')
# net = tflearn.dropout(net, 0.5, name='Dropout4')
net = tflearn.fully_connected(net, 1000, activation='relu', name='Layer5')
# net = tflearn.dropout(net, 0.5, name='Dropout5')

net = tflearn.fully_connected(net, 1, activation='linear')

net = tflearn.regression(net, optimizer='Adam', loss='mean_square', metric='accuracy', learning_rate=0.001)

# Define model
model = tflearn.DNN(net, tensorboard_verbose=3, tensorboard_dir="/tmp/tflearn_logs/test")

# Start training (apply gradient descent algorithm)
# print X
# print Y
earlyStoppingCallback = EarlyStoppingCallback(loss_threshold = 60)
model.fit(X, Y, n_epoch=10000, batch_size=60, show_metric=True,  snapshot_epoch=True,  validation_set=0.25, callbacks=earlyStoppingCallback,  run_id="tensorboard_test")

# Predictive Data('best')
estimation_result = model.predict(x_test)

# Results
result = np.concatenate((estimation_result,y_test),axis=1)
print('======================= DNN regression Results  ==================')
print('Estimation Result / Target ')
print(result)
print('==================================================================')
mape = mean_absolute_percentage_error(y_test, estimation_result)
print('Mean absolute percentage error :',mape)
# mse = mean_squared_error(y_test, estimation_result)
# print(mse)

# # Save a model
model.save('tensorboard_test2.tflearn')

Training Step: 7377  | total loss: 133.95444 | time: 0.787s
| Adam | epoch: 1054 | loss: 133.95444 - binary_acc: 0.0000 -- iter: 360/390
Training Step: 7378  | total loss: 132.92976 | time: 1.935s
| Adam | epoch: 1054 | loss: 132.92976 - binary_acc: 0.0000 | val_loss: 76.89469 - val_acc: 0.0000 -- iter: 390/390
